src/Features/Core/Portable/EmbeddedLanguages/EmbeddedLanguageDetector.cs

```diff
@@ -15,11 +15,12 @@ namespace Microsoft.CodeAnalysis.EmbeddedLanguages
 {
     internal readonly struct EmbeddedLanguageDetector(
         EmbeddedLanguageInfo info,
-        ImmutableArray<string> languageIdentifiers)
+        ImmutableArray<string> languageIdentifiers,
+        EmbeddedLanguageCommentDetector commentDetector)
     {
         private readonly EmbeddedLanguageInfo Info = info;
-        private readonly HashSet<string> LanguageIdentifiers = new HashSet<string>(languageIdentifiers, StringComparer.OrdinalIgnoreCase);
-        private readonly EmbeddedLanguageCommentDetector _commentDetector = new EmbeddedLanguageCommentDetector(languageIdentifiers);
+        private readonly HashSet<string> LanguageIdentifiers = new(languageIdentifiers, StringComparer.OrdinalIgnoreCase);
+        private readonly EmbeddedLanguageCommentDetector _commentDetector = commentDetector;
```

Comment:
instead of creating a fresh one each time, we are passed the cached singleton to use.